src/with_clause_parser.c

```diff
@@ -113,6 +114,24 @@ ts_with_clauses_parse(const List *def_elems, const WithClauseDefinition *args, S
 	return results;
 }
 
+extern TSDLLEXPORT char *
+ts_with_clause_result_deparse_value(const WithClauseResult *result)
+{
+	Oid oid = result->definition->type_id;
+	if (!OidIsValid(oid))
+		elog(ERROR, "argument \"%d\" has invalid OID", oid);
```

Comment:
`elog(ERROR ...)` is for internal program errors, like an assertion failure, so we probably don't want to be able to trigger it. If it's a user error, it should use `ereport` instead.

Reply:
no; there is no way for the user to affect this - I don't expect it to go off - its here for just in case something really odd is happening....

I could probably:
* change it to an assertion instead
* force-it to fail from an api test
* or remove it

I think an assertion would be the most appropriate in this case

Reply:
> no; there is no way for the user to affect this - I don't expect it to go off - its here for just in case something really odd is happening....

Ok. Makes sense to not have a test for this then.

You might want to consider using `Ensure` here, which is intended for this usage. It will abort execution in debug build if the condition does not hold, which makes it easier to debug since you will have a core dump. In release builds, it will just be an error message, just like here.

> or remove it

I do not think that we should remove such checks. It will just lead to crashes in production code, which causes a lot of other problems since it brings the entire server down. Better to print an error and abort if the impossible, or less plausible, happens.

Reply:
replaced it with `Ensure` - that's even better than the original check. I usually try to retain these kind of checks as much as possible.